crates/bevy_asset/src/loader.rs

```diff
@@ -235,20 +235,19 @@ impl_downcast!(AssetLifecycle);
 
 impl<T: AssetDynamic> AssetLifecycle for AssetLifecycleChannel<T> {
     fn create_asset(&self, id: HandleId, asset: Box<dyn AssetDynamic>, version: usize) {
-        if let Ok(asset) = asset.downcast::<T>() {
-            self.sender
-                .send(AssetLifecycleEvent::Create(AssetResult {
-                    asset,
-                    id,
-                    version,
-                }))
-                .unwrap();
-        } else {
+        let asset = asset.downcast::<T>().unwrap_or_else(|_| {
             panic!(
                 "Failed to downcast asset to {}.",
                 std::any::type_name::<T>()
-            );
-        }
+            )
+        });
```

Comment:
```rs
        let Some(asset) = asset.downcast::<T>() else {
            panic!(
                "Failed to downcast asset to {}.",
                std::any::type_name::<T>()
            )
        };
```
or even
```rs
        let asset = asset.downcast::<T>().expect(
            format!(
                "Failed to downcast asset to {}.",
                std::any::type_name::<T>()
            )
        );
```
but the latter may have performance impact of formatting even when we're not going to panic